dhis_android_analytics/src/test/java/dhis2/org/analytics/charts/providers/ChartCoordinatesProviderImplTest.kt

```diff
@@ -15,7 +16,9 @@ import org.mockito.Mockito
 class ChartCoordinatesProviderImplTest {
     private val d2: D2 = Mockito.mock(D2::class.java, Mockito.RETURNS_DEEP_STUBS)
     private val periodStepProvider: PeriodStepProvider = mock()
-    private val coordinatesProvider = ChartCoordinatesProviderImpl(d2, periodStepProvider)
+    private val resourceManager: ResourceManager = mock()
+    private val coordinatesProvider =
+        ChartCoordinatesProviderImpl(d2, periodStepProvider, resourceManager)
 
     @Test
     fun `Should get coordinates for data elements`() {
```

Comment:
These four tests are failing, producing a null pointer exception in `ChartCoordinatesProviderImpl line 47`. Please check it and run the tests.